crates/subspace-block-relay/src/protocol/compact_block.rs

```diff
@@ -329,7 +327,7 @@ where
                         "relay::on_request: missing entry not found"
                     );
                 }
-                Err(err) => return Err(format!("on_request: missing entry lookup: {err:?}").into()),
+                Err(err) => return Err(err),
```

Comment:
`?` can be used if you just forward error up, makes it a bit cleaner

Reply:
It is only 3-level when you match on `Result`, otherwise it could have been something like this:
```rust
            if let Some(ret) = self
                .backend
                .protocol_unit(&req.download_unit_id, &protocol_unit_id)?
            {
                protocol_units.insert(missing_id, ret);
            } else {
                warn!(
                    target: LOG_TARGET,
                    "relay::on_request: missing entry not found"
                );
            }
```